run/src/index.ts

```diff
@@ -86,22 +87,20 @@ export function setup<So extends Sources, Si extends FantasySinks<Si>>(
     );
   }
 
-  const sinkProxies = makeSinkProxies<So, Si>(drivers);
-  const sources = callDrivers<So, Si>(drivers, sinkProxies);
-  const adaptedSources = adaptSources(sources);
-  const sinks = main(adaptedSources);
+  const engine = setupReusable(drivers);
+  const sinks = main(engine.sources);
   if (typeof window !== 'undefined') {
     (window as any).Cyclejs = (window as any).Cyclejs || {};
     (window as any).Cyclejs.sinks = sinks;
   }
   function _run(): DisposeFunction {
-    const disposeReplication = replicateMany(sinks, sinkProxies);
+    const disposeRun = engine.run(sinks);
     return function dispose() {
-      disposeSources(sources);
-      disposeReplication();
+      disposeRun();
+      engine.dispose();
     };
   }
-  return {sinks, sources, run: _run};
+  return {sinks, sources: engine.sources, run: _run};
 }
```

Comment:
This is not crucial for the PR, but it rewrites `setup` in terms of `setupReusable`.